pinot-core/src/main/java/org/apache/pinot/core/data/manager/realtime/RealtimeTableDataManager.java

```diff
@@ -465,9 +465,7 @@ private void downloadSegmentFromDeepStore(String segmentName, IndexLoadingConfig
    */
   private void untarAndMoveSegment(String segmentName, IndexLoadingConfig indexLoadingConfig, File segmentTarFile)
       throws IOException {
-    // TODO: This could leave temporary directories in _indexDir if JVM shuts down before the temporary directory is
```

Comment:
maybe keep this comment (removing the `TODO:`) for the context, and add how we cleanup the tmp dirs now.